adapters/wazirx-nft-Xauction-bsc/index.js

```diff
@@ -0,0 +1,176 @@
+require("dotenv").config();
+
+const web3 = require("web3");
+const moment = require("moment");
+const BigNumber = require("bignumber.js");
+const Ethereum = require("../../sdk/EVMC");
+const axios = require("axios");
+const URL = "http://nft-sales-service.dappradar.com/open-source";
+const TRANSFER_TOPIC = "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef";
+const KEY = process.env.DAPPRADAR_API_KEY;
+const path = require("path");
+const { AVAILABLE_PROTOCOLS } = require("../../sdk/constants");
+
+class WazirxNft {
+    constructor() {
+        this.name = "WazirXNFT";
+        this.symbol = "WRXNFT";
+        this.currencyCode = {
+            "0": {
+                symbol: "WRX",
+                decimals: 8,
+                token_address: "0x8e17ed70334c87ece574c9d537bc153d8609e2a3",
+            },
+            "1": {
+                symbol: "WRX",
+                decimals: 8,
+                token_address: "0x8e17ed70334c87ece574c9d537bc153d8609e2a3",
+            },
+            "2": {
+                symbol: "BNB",
+                decimals: 18,
+                token_address: "bnb",
+            },
+        };
+        this.token = "0x0000000000000000000000000000000000000000";
+        this.protocol = "bsc";
+        this.token_symbol = "bnb";
+        this.block = 14571030;
+        this.nftContract = "0x23Cad0003e3A2b27b12359B25c25dD9a890AF8e1";
+        this.contract = "0x12E2EA4ADb0C384FB273279c9E169E2f494CcA21";
+        this.events = ["AuctionFinalized"];
+        this.pathToAbi = path.join(__dirname, "./abi.json");
+        this.range = 500;
+        this.chunkSize = 6;
+        this.sdk = new Ethereum(this);
+    }
+
+    run = async () => {
+        this.sdk = this.loadSdk();
+        await this.sdk.run();
+    };
+
+    loadSdk = () => {
+        return new Ethereum(this);
+    };
+    getSymbol = async () => {
```

Comment:
Call this function like in example (https://github.com/dappradar/nft-sales-adapters/blob/0bde056e7b5cf78d653b764cbd1fd6c6331e1933/adapters/ens/index.js#L27). This helps to seek a single template between different providers.

Reply:
The reason why I am not using getSymbol is that the API fails to give symbol data.

tokenAddress: 0x23Cad0003e3A2b27b12359B25c25dD9a890AF8e1
token: [WazirXNFT (WRXNFT)]

Reply:
We've fixed the API, so it should work now.

Reply:
http://nft-sales-service.dappradar.com/open-source/token-metadata?key=[KEY]&token_address=0x23Cad0003e3A2b27b12359B25c25dD9a890AF8e1&protocol=bsc

The above request returns the empty symbol

Reply:
I think the issue here is that your token's smart contract is a proxy, so it might not get metadata from the contract. I'll ask my colleagues to add this metadata manually and get back to you asap.

Reply:
@liutaurasjanuskas Please let me know if there is any update for me ?

Reply:
Hello. You need to provide an ERC20 token here. You are trying to get metadata of ERC721 or ERC1155.
For example, you can use these values: `eth`, `bnb`, `0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2` (wETH on Ethereum network), etc...

Reply:
Thanks, @liutaurasjanuskas .  So can you please confirm what does the token and token_symbol in the sale object refers to?

Reply:
Hello @puneethwrx, `token` = payment token and `token_symbol` = symbol of payment token. Is it clear enough now?

Reply:
Thanks a lot. Its clear now